x-pack/filebeat/input/s3/input.go

```diff
@@ -433,6 +434,17 @@ func (p *s3Input) newS3BucketReader(svc s3iface.ClientAPI, s3Info s3Info) (*bufi
 	if resp.Body == nil {
 		return nil, errors.New("s3 get object response body is empty")
 	}
+
+	if p.config.GZip && strings.HasSuffix(s3Info.key, ".gz") {
```

Comment:
Is it needed to add the GZip config parameter? Or can we just rely on checking the `.gz` suffix?

Reply:
Honestly, no. Logstash S3 input does gzip decompression seamlessly as well. I only added the parameter in case some users are shipping gzip as an event and then decompressing it somewhere else. That seems to be a corner case though and as long as the new feature is mentioned in the changelog, it shouldn't be an issue.

Reply:
Hmm I think I actually prefer without `GZip` config parameter (based on Logstash s3 input behavior). If user wants to ship gzip and decompress later, I haven't test this but I assume the s3 input `handleS3Objects` function will probably fail since it's looking for `reader.ReadString('\n')`.

Reply:
I agree. I will remove that parameter.